staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go

```diff
@@ -131,13 +132,19 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 		GenericAPIServer: genericServer,
 	}
 
-	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(apiextensions.GroupName, registry, Scheme, metav1.ParameterCodec, Codecs)
-	apiGroupInfo.GroupMeta.GroupVersion = v1beta1.SchemeGroupVersion
-	customResourceDefintionStorage := customresourcedefinition.NewREST(Scheme, c.GenericConfig.RESTOptionsGetter)
-	v1beta1storage := map[string]rest.Storage{}
-	v1beta1storage["customresourcedefinitions"] = customResourceDefintionStorage
-	v1beta1storage["customresourcedefinitions/status"] = customresourcedefinition.NewStatusREST(Scheme, customResourceDefintionStorage)
-	apiGroupInfo.VersionedResourcesStorageMap["v1beta1"] = v1beta1storage
+	apiResourceConfig := c.GenericConfig.MergedResourceConfig
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(apiextensions.GroupName, Registry, Scheme, metav1.ParameterCodec, Codecs)
+	if apiResourceConfig.AnyResourcesForVersionEnabled(v1beta1.SchemeGroupVersion) {
```

Comment:
@sttts so I'm guessing this is why you added it.  What if we controlled the addition of this delegation target inside of the kube-apiserver which is doing the composition?  Then the individual API servers (the vast majority of things that will exist) don't end up with logic blocks like this.

Reply:
It's optional anyway. The apiservers which do not care about runtime config (and don't add the APIEnablementOptions) will not have to use this. The kube-apiserver also opts-in to pass a runtime config to the apiextension-apiserver that comes from its flags. I think the actual logic belongs here. We could decide whether we want APIEnablementOptions by default in the standalone version.

Reply:
> t's optional anyway. The apiservers which do not care about runtime config (and don't add the APIEnablementOptions) will not have to use this. The kube-apiserver also opts-in to pass a runtime config to the apiextension-apiserver that comes from its flags. I think the actual logic belongs here. We could decide whether we want APIEnablementOptions by default in the standalone version.

as long as the code properly handles the nil case (enable everything), then I'm ok with it.